### PR TITLE
Fix: remove maxLength property for taxId and expiration date fields

### PR DESCRIPTION
**Pull request by eVoloshchak**
### Details
This fix is a result of the audit of all of the forms that aimed to fix any inconsistencies with modifying user input.
As was [discussed](https://expensify.slack.com/archives/C01GTK53T8Q/p1649702197809359) in slack, we should `Keep maxLength for some inputs that are not formatted (eg: CVV, last 4 digits of SSN, US zip code)`. Which means, we have to remove the `maxLength` property for TaxID field on [CompanyStep](https://github.com/Expensify/App/blob/main/src/pages/ReimbursementAccount/CompanyStep.js) and for expiration date field on [Add Debit card flow](https://github.com/Expensify/App/blob/main/src/pages/settings/Payments/AddDebitCardPage.js). Additionally, we have to do the following on the [CompanyStep](https://github.com/Expensify/App/blob/main/src/pages/ReimbursementAccount/CompanyStep.js) page:
1. Change validation to strip all non-numeric characters and then validate that there are exactly 9 numbers
2. Strip all non-numeric characters from `taxID` on form submission
3. Change placeholder of the `taxID` field 

### Fixed Issues
https://github.com/Expensify/App/issues/7524

### Tests
1. Navigate to the add debit card page:
    - Open Settings
    - Payments
    - Add payment method 
    - Debit card

2. Verify that expiration date filed does not have max length limit
3. Navigate to the company step page:
    - Create a workspace if you don't have one
    - Open Settings
    - Select your workspace 
    - Connect bank account
    - Connect manually
    - Fill out form and press "Save & continue" to get to step 2
4. Verify that taxID field does not have max length limit
5. Verify that taxID field accepts non-numeric characters and validation is correct (no error if there are exactly 9 numbers)

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [ ] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps
1. Navigate to the add debit card page:
    - Open Settings
    - Payments
    - Add payment method 
    - Debit card

2. Verify that expiration date filed does not have max length limit
3. Navigate to the company step page:
    - Create a workspace if you don't have one
    - Open Settings
    - Select your workspace 
    - Connect bank account
    - Connect manually
    - Fill out form and press "Save & continue" to get to step 2
4. Verify that taxID field does not have max length limit
5. Verify that taxID field accepts non-numeric characters and validation is correct (no error if there are exactly 9 numbers)

#### Web

https://user-images.githubusercontent.com/9059945/165086807-94620f45-9424-49cb-bd82-39e774164e82.mp4

https://user-images.githubusercontent.com/9059945/165086877-ba3beb03-bcb9-4d3b-a6a8-755a68d1d962.mp4

#### Mobile Web

https://user-images.githubusercontent.com/9059945/165087347-f6b7ddc2-4958-43c9-bcd2-34fd4680e655.mp4

https://user-images.githubusercontent.com/9059945/165087580-5430a255-2183-4bdf-bad9-d07a77e40c33.mp4

#### Desktop

I'm unable to launch the desktop app due to [this](https://expensify.slack.com/archives/C01GTK53T8Q/p1650494818552169?thread_ts=1650471398.505469&cid=C01GTK53T8Q) bug. It is not present if you downgrade to certain `node` and `npm` versions, but since I use MacinCloud, I'm unable to do so. So if someone with a mac could test it and upload a video, I would highly appreciate it.

#### iOS

https://user-images.githubusercontent.com/9059945/165088449-ae1f38a5-d37f-47e4-bf27-b583e768fb2c.mp4

https://user-images.githubusercontent.com/9059945/165088497-7eeca216-78eb-43ce-a243-d5d654dc5b8e.mp4

#### Android

https://user-images.githubusercontent.com/9059945/165088620-9b0111ab-028b-43c7-80d9-a46e95c902fc.mp4

https://user-images.githubusercontent.com/9059945/165088684-2f106585-4205-4afa-8d34-859f95e24786.mp4

### CODE REVIEW

**Comment by rushatgabhane** [comment]
> Navigate to the add debit card page


> Navigate to the company step page


@eVoloshchak please overexplain the steps such that someone unfamiliar with the app can also follow them, thanks!

File: src/pages/ReimbursementAccount/CompanyStep.js [hunk 1]
```diff
@@ -142,7 +142,7 @@ class CompanyStep extends React.Component {
             errors.website = true;
         }
 
-        if (!/[0-9]{9}/.test(this.state.companyTaxID)) {
+        if (!/^\d{9}$/.test(this.state.companyTaxID.replace(/\D/g, ''))) {
```
Comment (rushatgabhane) [inline src/pages/ReimbursementAccount/CompanyStep.js:145]: @eVoloshchak let's move this logic to `ValidationUtils`, and `CONST`
What do you say?

File: src/pages/ReimbursementAccount/CompanyStep.js [hunk 2]
```diff
@@ -176,7 +176,7 @@ class CompanyStep extends React.Component {
         }
 
         const incorporationDate = moment(this.state.incorporationDate).format(CONST.DATE.MOMENT_FORMAT_STRING);
-        BankAccounts.setupWithdrawalAccount({...this.state, incorporationDate});
+        BankAccounts.setupWithdrawalAccount({...this.state, incorporationDate, companyTaxID: this.state.companyTaxID.replace(/\D/g, '')});
```
Comment (rushatgabhane) [inline src/pages/ReimbursementAccount/CompanyStep.js:179]: @eVoloshchak we can proly move this regex to CONST too.

File: src/libs/ValidationUtils.js [hunk 3]
```diff
@@ -355,6 +355,16 @@ function isExistingRoomName(roomName, reports, policyID) {
...
 }
 
+/**
+ * Validate that this is a valid tax ID (consisting of 9 digits)
+ *
+ * @param {String} taxID
+ * @returns {Boolean}
+ */
+function isValidTaxID(taxID) {
+    return CONST.REGEX.TAX_ID.test(taxID.replace(/\D/g, ''));
```
Comment (rushatgabhane) [inline src/libs/ValidationUtils.js:365]: this one too

**Review by rushatgabhane — CHANGES_REQUESTED** [review]
ignore this

File: src/libs/ValidationUtils.js [hunk 4]
```diff
@@ -355,6 +355,16 @@ function isExistingRoomName(roomName, reports, policyID) {
     );
 }
 
+/**
+ * Validate that this is a valid tax ID (consisting of 9 digits)
```
Comment (rushatgabhane) [inline src/libs/ValidationUtils.js:359]:
```suggestion
 * Checks if tax ID consists of 9 digits
```

File: src/pages/ReimbursementAccount/CompanyStep.js (hunk 2 above)
Comment (rushatgabhane) [inline src/pages/ReimbursementAccount/CompanyStep.js:179]: NAB: @eVoloshchak we could move this regex to const too

File: src/libs/ValidationUtils.js (hunk 3 above)
Comment (rushatgabhane) [inline src/libs/ValidationUtils.js:365]: here too

**Review by rushatgabhane — CHANGES_REQUESTED** [review]

File: src/pages/ReimbursementAccount/CompanyStep.js (hunk 2 above)
Comment (eVoloshchak) [reply src/pages/ReimbursementAccount/CompanyStep.js]: True, updated the PR

**Review by rushatgabhane — APPROVED** [review]
@deetergp Looks good, and tests well.


<details>
<summary><h4> #### PR Reviewer Checklist</h4></summary>

- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

**Review by deetergp — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**